src/uct/ib/base/ib_device.c

```diff
@@ -1475,6 +1513,58 @@ uct_ib_device_get_roce_ndev_name(uct_ib_device_t *dev, uint8_t port_num,
     return UCS_OK;
 }
 
+ucs_status_t
+uct_ib_device_get_roce_ndev_index(uct_ib_device_t *dev, uint8_t port_num,
+                                  int gid_index, int *ndev_index_p)
+{
+    uct_ib_device_to_ndev_key_t ib_dev = {.dev = dev,
+                                          .port_num = port_num,
+                                          .gid_index = gid_index};
+    ucs_status_t status;
+    uct_ib_device_to_ndev_val_t *ndev_p;
+    khiter_t iter;
+    int khret;
+
+    pthread_mutex_lock(&uct_ib_device_to_ndev_cache_lock);
+    iter = kh_get(uct_ib_device_to_ndev, &ib_dev_to_ndev_map, ib_dev);
+    if (ucs_likely(iter != kh_end(&ib_dev_to_ndev_map))) {
+        ndev_p         = &kh_val(&ib_dev_to_ndev_map, iter);
+        *ndev_index_p  = ndev_p->if_index;
+        status         = UCS_OK;
+        goto out_unlock;
+    }
+
+    iter = kh_put(uct_ib_device_to_ndev, &ib_dev_to_ndev_map, ib_dev, &khret);
```

Comment:
since this is not fast path we can do kh_put only, and examine the return value khret to see if the rule exists